csharp/src/SeedLang.Block/Program.cs

```diff
@@ -22,27 +23,26 @@ namespace SeedLang.Block {
   // JSON files then wrapped into a ZIP package.
   //
   // This is a singleton class. There can be only one program instance in the memory.
-  public class BlockProgram {
-    private static readonly object _padlock = new object();
-    private static BlockProgram _instance = null;
+  public sealed class Program {
+    private static readonly Lazy<Program> _engine = new Lazy<Program>(() => new Program());

```

Comment:
The name _engine shall be _lazyInstance or something else?

Reply:
Done. The name "engine" was copied from SeedLang.Engine. Forgot to update it.